src/backend/src/signer.rs

```diff
@@ -0,0 +1,60 @@
+//! Code for inetracting with the chain fusion signer.
+use crate::state::{CYCLES_LEDGER, SIGNER};
+use candid::{CandidType, Deserialize, Nat, Principal};
+use ic_cycles_ledger_client::{Account, ApproveArgs, ApproveError, Service as CyclesLedgerService};
+use ic_ledger_types::Subaccount;
+use serde_bytes::ByteBuf;
+
+#[derive(CandidType, Deserialize, Debug, Clone, Eq, PartialEq)]
+pub enum AllowSigningError {
+    Other(String),
+    FailedToContactCyclesLedger,
+    ApproveError(ApproveError),
+}
+
+/// Enables the user to sign transactions.
+///
+/// Signing costs cycles.  Managing that cycle payment can be painful so we take care of that.
+pub async fn allow_signing() -> Result<(), AllowSigningError> {
+    let cycles_ledger: Principal = *CYCLES_LEDGER;
+    let signer: Principal = *SIGNER;
+    let caller = ic_cdk::caller();
+    let expected_fee = 1_000_000_000u64;
+    let amount = Nat::from(100 * expected_fee);
```

Comment:
Why 100? Why multiplied the expected fee, is the expected_fee the conversion to TCycles?

Reply:
Updated the signer fee to the value it will most likely be when th ecfs goes live.

Updated the number of ops to a new arbitrary value of 30, in view of how expensive the operations are.